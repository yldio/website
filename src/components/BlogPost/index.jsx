import PropTypes from 'prop-types';
import React, { Fragment } from 'react';
import Markdown from 'react-markdown';
import styled from 'styled-components';

const Title = styled.h1`
  font-size: 3em;
`;

const BlogPost = ({ title, content, authors }) => (
  <Fragment>
    <Title>{title}</Title>
    <Markdown source={content} />
    <p>{authors.map(author => author.name)}</p>
  </Fragment>
);

BlogPost.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
  authors: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default BlogPost;

export const fragment = graphql`
  fragment BlogPostFrontmatter on Posts {
    title
    content
    authors {
      name
      avatar {
        url
      }
    }
  }
`;
