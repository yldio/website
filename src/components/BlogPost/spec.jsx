import React from 'react';
import { shallow } from 'enzyme';
import BlogPost from './index';

test('components/BlogPost snapshot', () => {
  const authors = [{ name: 'Nuno' }];

  const wrapper = shallow(
    <BlogPost
      title="New YLD Blog"
      content="Some markdown content..."
      authors={authors}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
