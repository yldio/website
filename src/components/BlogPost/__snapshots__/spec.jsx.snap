// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/BlogPost snapshot 1`] = `
<React.Fragment>
  <styled.h1>
    New YLD Blog
  </styled.h1>
  <ReactMarkdown
    astPlugins={Array []}
    escapeHtml={true}
    plugins={Array []}
    rawSourcePos={false}
    renderers={Object {}}
    skipHtml={false}
    source="Some markdown content..."
    sourcePos={false}
    transformLinkUri={[Function]}
  />
  <p>
    Nuno
  </p>
</React.Fragment>
`;
